src/autumnaton.ts

```diff
@@ -0,0 +1,224 @@
+    const acuityCutoff = 20 - (acuityOverride ?? AutumnAton.visualAcuity()) * 5;
+    const validDrops = monsters
+      .map((m) => itemDropsArray(m))
+      .flat()
+      .map(({ rate, type, drop }) => ({
+        value: !["c", "0"].includes(type) ? garboValue(drop, true) : 0,
+        preAcuityExpectation: ["c", "0", ""].includes(type) ? (2 * rate) / 100 : 0,
+        postAcuityExpectation:
+          rate >= acuityCutoff && ["c", "0", ""].includes(type) ? (8 * rate) / 100 : 0,
+      }));
+    const overallExpectedDropQuantity = sum(
+      validDrops,
+      ({ preAcuityExpectation, postAcuityExpectation }) =>
+        preAcuityExpectation + postAcuityExpectation
+    );
+    const expectedCollectionValue = sum(
+      validDrops,
+      ({ value, preAcuityExpectation, postAcuityExpectation }) => {
+        // This gives us the adjusted amount to fit within our total amount of available drop slots
+        const adjustedDropAmount =
+          (preAcuityExpectation + postAcuityExpectation) *
+          Math.min(1, maximumDrops / overallExpectedDropQuantity);
+        return adjustedDropAmount * value;
+      }
+    );
+    return seasonalItemValue(location) + expectedCollectionValue;
+  }
+}
+
+function seasonalItemValue(location: Location, seasonalOverride?: number): number {
+  // Find the value of the drops based on zone difficulty/type
+  const autumnItems = $items`autumn leaf, AutumnFest ale, autumn breeze, autumn dollar, autumn years wisdom`;
+  const avgValueOfRandomAutumnItem = garboAverageValue(...autumnItems);
+  const autumnMeltables = $items`autumn debris shield, autumn leaf pendant, autumn sweater-weather sweater`;
+  const autumnItem = getAutumnatonUniques(location)[1];
+  const seasonalItemDrops = seasonalOverride ?? AutumnAton.seasonalItems();
+  return (
+    (seasonalItemDrops > 1 ? avgValueOfRandomAutumnItem : 0) +
+    (autumnMeltables.includes(autumnItem)
+      ? // If we already have the meltable, then we get a random item, else value at 0
+        availableAmount(autumnItem) > 0
+        ? avgValueOfRandomAutumnItem
+        : 0
+      : garboValue(autumnItem, true))
+  );
+}
+
+function expectedRemainingExpeditions(legOverride?: number): number {
+  const availableAutumnatonTurns = estimatedTurns() - AutumnAton.turnsLeft();
+  let expeditionTurnSum = 0;
+  let quests = get("_autumnatonQuests", 0);
+  while (expeditionTurnSum < availableAutumnatonTurns) {
+    expeditionTurnSum +=
+      11 *
+      Math.max(
+        1,
+        quests -
+          (legOverride ?? AutumnAton.currentUpgrades().filter((u) => u.includes("leg")).length)
+      );
+    quests++;
+  }
+
+  return quests - get("_autumnatonQuests", 0);
+}
+
+const profitRelevantUpgrades = [
+  "leftarm1",
+  "leftleg1",
+  "rightarm1",
+  "rightleg1",
+  "cowcatcher",
+  "periscope",
+  "radardish",
+] as const;
+
+export function prioritizeUpgradeLocations(fullLocations: Location[]): Location[] {
```

Comment:
I think `prioritize` is a misleading name here--it sounds to me like it sorts the list, as opposed to filtering it

Reply:
The best I can think of atm is `trimToBestUpgradeLocations` what do you think?